src/clj/clj_pdf/core.clj

```diff
@@ -1044,34 +1056,35 @@
     (write-total-pages width doc-meta temp-stream output-stream)))
 
 (defn- to-pdf [input-reader r out]
-  (let [doc-meta (input-reader r)
-        [^Document doc
-         width height
-         ^ByteArrayOutputStream temp-stream
-         ^OutputStream output-stream
-         ^PdfWriter pdf-writer] (setup-doc doc-meta out)]
+  (let [doc-meta (input-reader r)]
     (register-fonts doc-meta)
-    (loop []
-      (if-let [item (input-reader r)]
-        (do
-          (add-item item doc-meta width height doc pdf-writer)
-          (recur))
-        (do
-          (.close doc)
-          (write-total-pages width doc-meta temp-stream output-stream))))))
+    (let [[^Document doc
+           width height
+           ^ByteArrayOutputStream temp-stream
+           ^OutputStream output-stream
+           ^PdfWriter pdf-writer] (setup-doc doc-meta out)]
+      (loop []
+        (if-let [item (input-reader r)]
+          (do
+            (add-item item doc-meta width height doc pdf-writer)
+            (recur))
+          (do
+            (.close doc)
+            (write-total-pages width doc-meta temp-stream output-stream)))))))
 
 (defn- seq-to-doc [items out]
-  (let [doc-meta (first items)
-        [^Document doc
-         width height
-         ^ByteArrayOutputStream temp-stream
-         ^OutputStream output-stream
-         ^PdfWriter pdf-writer] (setup-doc doc-meta out)]
+  (let [doc-meta (first items)]
```

Comment:
Might be better to pull this out into a function, something like `(parse-meta (first items))`. That could run `register-fonts` and return the `doc-meta`. That would avoid the need for nested `let` statements.